odd-platform-ui/src/components/Management/DataSourcesList/DataSourceItem/DataSourceItemToken/DataSourceItemToken.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import {
+  DataSource,
+  DataSourceApiRegenerateDataSourceTokenRequest,
+} from 'generated-sources';
+import ConfirmationDialog from 'components/shared/ConfirmationDialog/ConfirmationDialog';
+import { Typography } from '@mui/material';
+import AppButton from 'components/shared/AppButton/AppButton';
+import CopyButton from 'components/shared/CopyButton/CopyButton';
+import { TokenContainer, Token } from './DataSourceItemTokenStyles';
+
+interface DataSourceItemProps {
+  dataSource: DataSource;
+  regenerateDataSourceToken: (
+    params: DataSourceApiRegenerateDataSourceTokenRequest
+  ) => Promise<DataSource>;
+}
+
+const DataSourceItemToken: React.FC<DataSourceItemProps> = ({
+  dataSource,
+  regenerateDataSourceToken,
+}) => {
+  const checkTokenLength = (token: string): boolean =>
```

Comment:
You're not checking it's length, in fact you're checking if it is hidden/masked (contains asterisks).
```suggestion
  const checkIsTokenMasked = (token: string): boolean =>
```